File: Projeto_4/tests/teste.cpp
#include <iostream>
#include <cassert>
#include "../src/Aluno.h"
#include "../src/Disciplina.h"

using namespace std;

void testarDisciplina() {
    Disciplina d("História", 40, 7.0);
    assert(d.getNome() == "História");
    assert(d.getNota() == 7.0);
    assert(verificarAprovacao(d) == true);

    d.setNota(5.5);
    assert(d.getNota() == 5.5);
    assert(verificarAprovacao(d) == false);
    cout << "Teste de Disciplina passou.\n";
}

void testarAluno() {
    Aluno a("Carlos", "202500001", "Direito");
    a.exibirInformacoes(); // teste visual
    cout << "Teste de Aluno (visual) exibido com sucesso.\n";
}

int main() {
    testarDisciplina();
    testarAluno();
    cout << "Todos os testes foram executados.\n";
    return 0;
}

File: Projeto_1/PROJETO_1 - MarioAlex/tests/teste.py
from src.cliente import Cliente

def test_cliente():
    cliente = Cliente("Teste", 25, 1000.0)
    assert cliente.get_nome() == "Teste"
    assert cliente.get_idade() == 25
    assert cliente.get_saldo() == 1000.0

    cliente.atualizar_saldo(500.0)
    assert cliente.get_saldo() == 1500.0
